storage/factory/pruningStorerFactory.go

```diff
@@ -250,12 +250,14 @@ func (psf *StorageServiceFactory) CreateForShard() (dataRetriever.StorageService
 	store.AddStorer(dataRetriever.ReceiptsUnit, receiptsUnit)
 	store.AddStorer(dataRetriever.TrieEpochRootHashUnit, trieEpochRootHashStorageUnit)
 
-	err = psf.setupDbLookupExtensions(store, &successfullyCreatedStorers)
+	createdStorers, err := psf.setupDbLookupExtensions(store)
+	successfullyCreatedStorers = append(successfullyCreatedStorers, createdStorers...)
```

Comment:
maybe move the `append` sections after the error check? it's useless to append to a slice if you return an error so the slice you have just appended to is never used

Reply:
Won't do. Please check the whole function. The `successfullyCreatedStorers` is used in the defer function as to cleanup the already-opened storers. It should stay as it is, otherwise the defer function will not close the storers that could have been opened by one of those 2 functions.